vite-project-kabinet/src/Content/Content.jsx

```diff
@@ -1,20 +1,32 @@
 import "./content.style.css"
 import { PaginatedList } from "./PaginatedList.jsx"
 import { LoadedList } from "./LoadedList.jsx"
+import { useState, useEffect } from "react"
 
 
 export const Content = ({articles}) => {
 
+  const [viewportWidth, setViewportWidth] = useState(window.innerWidth)
+
+  useEffect(() => {
+
+    const handleResize = () => {
+      setViewportWidth(window.innerWidth)
+    }
+
+    window.addEventListener("resize", handleResize)
+    return () => {
+      window.removeEventListener("resize", handleResize)
+    }
```

Comment:
Tohle je dobře. Rozumíš tomu, co to dělá?

Reply:
A jak jsi se ptala, na to použití `useEffect`... sama jsi psala, že chápeš použití když se něco má použít jen jednou. Jak jinak bys vyřešila přidání eventListenera, což chceš udělat jen jednou na komponentu, bez useEffect hooku?

Reply:
Chápem čo to robí, len mi nebolo jasné, že prečo je doležité ošetriť to, aby sa tam eventListener pridal len raz. Rozumiem tomu, že duplicitné listenery nechcem. Ale asi som si nevedela predstaviť situáciu, kedy sa moja komponenta Content vykreslí viac než raz.